lib/Sema/TypeCheckType.cpp

```diff
@@ -3987,6 +3987,14 @@ TypeResolver::resolveExistentialType(ExistentialTypeRepr *repr,
   if (constraintType->hasError())
     return ErrorType::get(getASTContext());
 
+  // Diagnose redundant `any` on an already existential type e.g. any (any P)
+  // with a fix-it to remove first any.
+  if (constraintType->is<ExistentialType>()) {
+    diagnose(repr->getLoc(), diag::redundant_any_in_existential, constraintType)
+        .fixItRemove(repr->getAnyLoc());
+    return constraintType;
+  }
+
   if (!constraintType->isExistentialType()) {
```

Comment:
Can we instead fix this here by changing this to `(!constraintType->isConstraintType())`? That seems like the obvious fix to me

Reply:
`isConstraintType()` return `true` for `ProtocolType`, `ProtocolCompositionType`, and `ParameterizedProtocolType`, but not `ExistentialType`. This also matches the assertion inside `ExistentialType::get`

Reply:
It makes sense, just adjusted!

Reply:
Sorry @LucianoPAlmeida , I'm completely useless for this one :)

Reply:
Ah no worries =]